test/future.test.js

```diff
@@ -191,6 +206,40 @@ describe('Future', function() {
 
   });
 
+  describe('#concat', function() {
+
+    it('resolves as soon as the first Future settles', function(done){
+      this.timeout(25);
+      delayValue(15, 1).concat(delayValue(100, 2)).fork(fail(done), assertCbVal(done, 1));
+    });
+
+    it('rejects if the first to settle rejects', function(done){
+      this.timeout(25);
+      delayError(15, 1).concat(delayValue(100, 2)).fork(assertCbVal(done, 1), fail(done));
+    });
+
+    it('associativity is not broken by race-conditions', function(done){
+      var i, assoc = types.semigroup.associative;
+
+      var randomDelayValue = function(value){
+        return Future(function(reject, resolve){
+          setTimeout(resolve, 5 + Math.round(Math.random() * 20), value);
+        });
+      };
+
+      var f1 = randomDelayValue(1);
+      var f2 = randomDelayValue(2);
+      var f3 = randomDelayValue(3);
+
+      for(i=0; i<20; i++){
+        assert.equal(true, assoc(f1, f2, f3));
+      }
+
+      setTimeout(done, 50);
```

Comment:
Because of the way `Future.prototype.equals` is implemented, and the way I need to perform my test several times to be sure it passes (the condition has to be random), the failure of this unit test will bleed into others. When the first `assoc()` test throws, the unit test will fail and Mocha will move on to the next. But there will usually still be some `assoc()` tests running in the background, which then also throw while Mocha is running some other test, therefore making it think those are failing too.

Shouldn't be a problem once this test passes, or is removed. But something to keep in mind if you decide to play around with this.